mdtraj/geometry/rdf.py

```diff
@@ -158,21 +158,31 @@ def compute_rdf_t(traj, pairs, times, period_length=None, r_range=None,
         pairs_set = np.unique(pairs)
         pairs = np.vstack([np.vstack([pairs_set, pairs_set]).T, pairs])
 
-    g_r_t = np.zeros(shape=(len(times), n_bins))
-    num_chunks = int(np.floor(traj.n_frames / period_length))
-
-    # Returns shape (len(times), len(pairs))
-    frame_distances = compute_distances_t(traj, pairs, times, periodic=periodic, opt=opt)
-
-    for n, distances in enumerate(frame_distances):
-        tmp, edges = np.histogram(distances, range=r_range, bins=n_bins)
-        g_r_t[n, :] += tmp
-    r = 0.5 * (edges[1:] + edges[:-1])
-
-    # Normalize by volume of the spherical shell (see above)
-    V = (4 / 3) * np.pi * (np.power(edges[1:], 3) - np.power(edges[:-1], 3))
-    norm = len(pairs) / (period_length) * np.sum(1.0 / traj.unitcell_volumes) * V
-
-    g_r_t = g_r_t.astype(np.float64) / norm  # From int64.
-
-    return r, g_r_t
+    g_r_t = []
+    weights = []
+
+    # Splits pairs into smaller chunks so that frame_distances is not excesivly large
```

Comment:
I chose 100000 fairly arbitrarily based on a desired final size. I took 5 GB as the goal, then did the following logic:
8 bytes/element * num_concurrent_pairs * 200 times * 32 cpu processes = 5 GB. Solving gives num_concurrent_pairs ~= 100000. This number can likely be increased quite a bit from where it is now, I'll do some timing to see how sensitive the speed of the code is to it.

Reply:
After testing with `mprof`, 100000 seems to be the sweet spot for both memory usage and speed with 200 times analyzed and my test system of 2000 water molecules. It was actually faster than the original since my desktop did not need to buffer ram. I made it a flag so the user can change it if they wish.